Lib/xml/etree/ElementPath.py

```diff
@@ -281,12 +288,23 @@ def select(context, result):
                         if "".join(e.itertext()) == value:
                             yield elem
                             break
+            def select_negated(context, result):
+                for elem in result:
+                    for e in elem.findall(tag):
```

Comment:
I'd suggest using `iterfind()` here, since it breaks on the first match anyway.
```suggestion
                    for e in elem.iterfind(tag):
```

Reply:
Done, the regular `select` function for this branch also uses `findall` so might be worth making that consistent/cleaning up in a separate commit.